modules/bwa/mem/main.nf

```diff
@@ -20,6 +20,7 @@ process BWA_MEM {
     def args2 = task.ext.args2 ?: ''
     def prefix = task.ext.prefix ?: "${meta.id}"
     def read_group = meta.read_group ? "-R ${meta.read_group}" : ""
+    def split_cpus = Math.floor(task.cpus/2)
```

Comment:
we add issues with task.cpus because both bwa mem and samtools are requesting the same amount of cpus, so you end up eating more cpus than requested, we had a big discussion about it somewhere, and we ended up doing that in sarek

Reply:
hm here is some related discussion https://nfcore.slack.com/archives/CJRH30T6V/p1632822976224200
almost seems like it might be good to overthink this line as far as i understand

Reply:
in that case the documentation for multitool modules would need to be updated to

Reply:
Ping @pontus for a once over.

Reply:
Without actually having checked how they use these threads, the usage that makes most sense to me is to let the system handle things (i.e. `-@ task.cpus`).

There has (as noted by Friederike) been discussions about broken systems where the batch system kills jobs that uses too much CPU rather than restricting how much they can use to start with.

I believe those are very few, and it makes more sense to me to address in other ways if it happens (how would depend on what's going on).

It's also _theoretically_ possible that job performance degrades by telling it to use more threads (e.g. because of CPU cache exhaustion leading to stalls, limited memory bandwidth leading to stalls),  but a sensible basic approach is 
 to not limit tools if not needed and troubleshoot cases where that's visibly suboptimal. Modern operating systems are very good at handling thousands of threads in a good way.

Reply:
ok, then I remove all that, and we'll figure it out later on if we need that or not